CONTRIBUTING.md

```diff
@@ -0,0 +1,116 @@
+# Contributing guidelines
+
+## Before contributing
+
+Welcome to [TheAlgorithms/TypeScript](https://github.com/TheAlgorithms/TypeScript)! Before sending your pull requests,
+make sure that you **read the whole guidelines**. If you have any doubt on the contributing guide, please feel free to
+[state it clearly in an issue](https://github.com/TheAlgorithms/TypeScript/issues/new).
+
+## Contributing
+
+### Contributor
+
+We are very happy that you consider implementing algorithms and data structures for others! This repository is
+referenced and used by learners from around the globe. Being one of our contributors, you agree and confirm that:
+
+- You did your work - plagiarism is not allowed.
+  - Any plagiarized work will not be merged.
+- Your work will be distributed under [MIT License](LICENSE) once your pull request is merged.
+- Your submitted work must fulfill our styles and standards.
+
+**New implementation** is welcome! For example, new solutions to a problem, different representations of a graph data
+structure or algorithm designs with different complexity.
+
+**Improving comments** and **writing proper tests** are also highly welcome.
+
+### Contribution
+
+We appreciate any contribution, from fixing grammar mistakes to implementing complex algorithms. Please read this
+section if you are contributing to your work.
+
+If you submit a pull request that resolves an open issue, please help us to keep our issue list small by adding
+`fixes: #{$ISSUE_NO}` to your commit message. GitHub will use this tag to auto-close the issue if your PR is merged.
+
+#### What is an Algorithm?
+
+An Algorithm is one or more functions (or classes) that:
+
+- take one or more inputs.
+- perform some internal calculations or data manipulations.
+- return one or more outputs.
+- have minimal side effects.
+
+Algorithms should be packaged in a way that would make it easy for readers to put them into larger programs.
+
+Algorithms should:
+
+- have intuitive class and function names that make their purpose clear to readers.
+- use TypScript naming conventions and intuitive variable names to ease comprehension.
+- be flexible to take different input values.
+- raise TypeScript exceptions (RangeError, etc.) on erroneous input values.
+
+Algorithms in this repo should not be how-to examples for existing TypeScript packages. Instead, they should perform
+internal calculations or manipulations to convert input values into different output values. Those calculations or
+manipulations can use data types, classes, or functions of existing TypeScript packages but each algorithm in this repo
+should add unique value.
+
+#### File Naming Convention
+
+- filenames should use the UpperCamelCase (PascalCase) style.
+- There should be no spaces in filenames.
+- **Example:** `UserProfile.ts` is allowed but `userprofile.ts`,`Userprofile.ts`,`user-Profile.ts`,`userProfile.ts` are
+  not.
+
```

Comment:
Should we include the use of semantic commit messages? These make it much more professional, IMO.

Reply:
I usually use the following format "<samp>commit type: short description</samp>" and put any extra information in the description. Some examples:
<samp>
feat: release candidate for v0.1.2
chore: fix spelling
fix: improper error message
</samp>